cookbook/mass_edition/register_a_new_mass_edit_action.rst

```diff
@@ -2,55 +2,111 @@ How to Register a New Mass Edit Action on Products
 ==================================================
 
 The Akeneo PIM comes with a number of mass edit actions.
-It also comes with a simple method to define your own mass edit action
+It also comes with a flexible way to define your own mass edit action
 on selected products.
 
-Creating a MassEditAction
--------------------------
-The first step is to create a new class that implements ``MassEditActionInterface`` or extends
-``ProductMassEditOperation`` or ``FamilyMassEditOperation`` (given on which grid you want to apply the operation):
 
-.. literalinclude:: ../../src/Acme/Bundle/EnrichBundle/MassEditAction/Operation/CapitalizeValues.php
-   :language: php
-   :prepend: # /src/Acme/Bundle/EnrichBundle/MassEditAction/Operation/CapitalizeValues.php
-   :linenos:
+Prerequisite
+------------
+The mass edit action uses the `BatchBundle
+<https://github.com/akeneo/BatchBundle/>`_ in order to run mass edit in background. Readers and Writers are already
+created so in this cookbook we will focus on how to create a Mass Edit Action and create a Processor.
+For more information on how to create a Job, Reader, Processor, or Writer please see 'create-specific-connector'
 
-.. literalinclude:: ../../src/Acme/Bundle/EnrichBundle/Form/Type/MassEditAction/CapitalizeValuesType.php
-  :language: php
-  :prepend: # /src/Acme/Bundle/EnrichBundle/Form/Type/MassEditAction/CapitalizeValuesType.php
-  :linenos:
+Work with a custom Acme bundle
+------------------------------
+Your custom Mass Edit actions have to be in a custom Acme bundle which inherit our ``EnrichBundle``.
+We advise you to create this custom bundle with the Symfony command:
+
+.. code-block:: bash
+
+  php app/console generate:bundle
```

Comment:
So you will generate routing, controllers etc. and classes you don't know just to create an empty bundle? Imho, the creation of a bundle should be in a specific part of the cookbook but we should not advise to create a custom bundle with the Sf command.
During training, we advise to create bundles by hand to understand what is done.